### PR TITLE
Documentation

### PR DESCRIPTION
**Pull request by T2heke**

### CODE REVIEW

**Comment by T2heke** [comment]
My part of the documentation is ready for review.

File: README.asciidoc [hunk 1]
```diff
@@ -4,7 +4,87 @@
...
+
+The code is broken down to 6 files::
+	* config_colors.py - Utility to configure colors. Check log for detailed instructions.
+	* color.py - contains color information and its enum class.
+	* camera.py - contains the camera proccessing code and describes methods required by the image processing module (for different camera types).
+	* image processor.py - main file for image processing, provides processed frame data. Object analysing and filtration.
+	* motion.py - Contains a IRobotMotion interface that descibes method required for moving the robot. Has the driving logic for omnimotion and sending data to the mainboard over serial connection. Also contains an example implementation (TurtleRobot) that visualizes motion using turtle tools in python.
+	* main.py - main logic file that contains the state machine for robot driving, basket states, referee server controlling and enables the controlling of the robot.
+
+The `main.py` holds the state machine for the driving logic. For configurations check the software branch readme file. The overall logic of the code is as shown on this https://drive.google.com/file/d/1yLGe8DUNGeWWGTZ_hRFPqS1l04g-p03I/view?usp=sharing[block diagram].
```
Comment (ReikoR) [inline README.asciidoc:46]: Add block diagram to the repository as a SVG image and add the image to the README.

**Review by ReikoR — CHANGES_REQUESTED** [review]

File: README.asciidoc [hunk 2]
```diff
@@ -52,25 +53,28 @@ I think the state machine could be in an another file as well as the referee ser
...
+
+The ball thrower was probably the trickiest of the parts, since when the thrower sucks, then it doesn't matter how good the code is as it will miss anyway. I probably overlooked the rigidity on my first few designs and it was clear that the accuracy would've suffered if I had't more spacers, deeper drilled holes etc. Also the area between the thrower motor and ground was pretty difficult to get right since you have to consider that the ball is soft and the bottom of the thrower can't slide against the ground. The easiest solution to that was to import an actual 3D model of the ball into the design and see how it passes through and where it gets stuck.
+
+
+
+For the construction of the parts I'd probably recommend coming up with some assisting tools for drilling the holes, maybe a thin 3D-printed sheet where the holes should be drilled. It's better to be safe than sorry since drilling in the holes a few mm could easily ruin your manufactured part.
+
+
+
+image:robot_CAD.PNG[]
```
Comment (ReikoR) [inline README.asciidoc:69]: Hide joints in Fusion 360 and create a new image.

File: README.asciidoc [hunk 3]
```diff
@@ -86,6 +90,19 @@ If I could suggest something to next year students then it is to pick your team
 
 If I could suggest something to the instructors it would to organize task presentations better. At the moment you come to class and wait, not knowing, if an instructor is going to be present nor if he/she has time for you. Maybe create a time-slot registration system?
 
+=== Kenno
+
+I worked on the mechanics part of the robot, whenever possible also helped with electronics when it was still being done.
+
+I learned that it is not all that difficult to create something as I first thought. Since I had used other CAD programs before this course then I also learned a thing or two about Fusion 360 and compare it's capabilites to other CAD programs. Bunch of new commands like rigid groups and circular patterns. Got to know how to do CAD for the first time and learn how to use the milling machine.
```
Comment (ReikoR) [inline README.asciidoc:97]: > Got to know how to do CAD

I'm guessing you mean "Got to know how to do CAM".

CAD -> CAM?

**Review by ReikoR — CHANGES_REQUESTED** [review]

**Review by ReikoR — APPROVED** [review]

**Pull request merged**